src/Bio.hs

```diff
@@ -0,0 +1,72 @@
+{-# LANGUAGE QuasiQuotes #-} 
+--{-# LANGUAGE OverlappingInstances #-} --lets override tuple
+
+import Data.Char(ord,chr,toUpper)
+import Control.Monad.Trans(liftIO)
+import Data.String.Here
+import System.Directory(doesFileExist)
+import Text.Regex(subRegex,mkRegex)
+import Control.Applicative
+import Control.Monad.Extra(ifM)
+
+data IUPAC = A | C | G | T | N
+  deriving (Eq, Ord, Show, Enum) --[A ..] == [A, C, G, T]
+
+type Id = String
+type Quality = [Int] 
+-- data FastaRecord = Rec Id DNASeq
+type DNASeq = [IUPAC] 
+data FastqRecord = Rec Id DNASeq Quality
+instance Show FastqRecord  where 
+     show (Rec id seq qual) = [i|@${id}
++
+${join $ map show seq}
+${map encodeQual qual}|] 
+       where join = foldr (++) "" 
+
+toIUPAC nt = [A ..] !! (length $ takeWhile (/= (toUpper nt)) "ACGTN")
+decodeQual = (subtract 33) . ord
+encodeQual = chr . (33 +)
+makeSeq (a:b:_:d:[]) = Rec (tail a) (map toIUPAC b) (map decodeQual d)
+
+splitEvery n = takeWhile (not . null) . map (take n) . iterate (drop n)
+
+readFastq :: FilePath -> IO [FastqRecord] 
+--e.g. fmap head $ readFastq "foo.fastq"
+readFastq p = do
+  str <- readFile p
```

Comment:
use lazy file reading.
also, I wonder if pattern-matching but ignoring evaluates the quality transformation to ints or the sequence transformation to [IUPAC]